joblib/_store_backends.py

```diff
@@ -185,12 +186,24 @@ def dump_item(self, path, item, verbose=1):
 
             def write_func(to_write, dest_filename):
                 with self._open_item(dest_filename, "wb") as f:
-                    numpy_pickle.dump(to_write, f,
-                                      compress=self.compress)
+                    try:
+                        numpy_pickle.dump(to_write, f, compress=self.compress)
+                    except PicklingError as e:
+                        warnings.warn(
+                            'Unable to cache to disk: '
+                            'failed to pickle output. '
+                            'In future versions of joblib '
+                            'this will raise an exception.'
+                            'Exception: %s.' % e,
```

Comment:
We try to avoid the old formating synthax in the new code base, so please use:
```suggestion
                            f'Exception: {e}.'
```

Reply:
Also, could you please rewrap the string to better use the available <80 char limit of PEP8?

Some people complain than 80 is not enough, others seem reluctant to use more than 60 apparently ;) 80 is a good middle ground.